packages/typography/src/Typography.tsx

```diff
@@ -117,33 +100,60 @@ Subtitle.displayName = 'Subtitle';
 /**
  * Body
  */
-type BodyProps = HTMLElementProps<'div'> & {
+
+type BodyFontWeight = 'regular' | 'medium';
+type BodyProps = HTMLElementProps<'p'> & {
   /**
    * font-weight applied to typography element
    * default: `regular`
    */
-  weight?: 'regular' | 'medium';
+  weight?: BodyFontWeight;
+  as?: keyof JSX.IntrinsicElements;
 };
 
-function Body({ children, className, weight = 'regular', ...rest }: BodyProps) {
+const Body: ExtendableBox<BodyProps, 'p'> = ({
+  className,
+  weight = 'regular',
+  as = 'p',
+  ...rest
+}: BodyProps) => {
   const size = useBaseFontSize();
   const body = size === 16 ? typeScale2 : typeScale1;
+  const fontWeights: {
+    [key: string]: {
+      [key in BodyFontWeight]: number;
+    };
+  } = {
+    default: {
+      regular: 400,
+      medium: 500,
+    },
+    strong: {
+      regular: 700,
+      medium: 800,
+    },
+    b: {
+      regular: 700,
+      medium: 800,
+    },
+  } as const;
+
+  const getFontWeight = () =>
+    as in fontWeights
+      ? fontWeights[as][weight]
+      : fontWeights['default'][weight];
 
   const fontWeight = css`
-    font-weight: ${weight === 'regular' ? 400 : 500};
-
-    strong,
-    b {
-      font-weight: ${weight === 'regular' ? 700 : 800};
-    }
```

Comment:
This wasn't really specced out anywhere, but it's just a minor pet peeve to see magic numbers, so I added this in. Hope that's alright :)

Reply:
Unless I'm missing something, the changes here aren't functionally the same. This update doesn't actually style descendent `strong` and `b` tags anymore, where it did before.

In general, I don't actually consider font weight numbers any more magic as using  `font-weight: bold` since the numbers have a very static definition in an application. I think what we previously had here is probably better given the intent and context of the code.

Reply:
I think I agree with Dave here, the original is a bit easier to read.
Here's the MDN article on `font-weight`. Using numbers here aren't magic, but actually correspond to weights https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight

Reply:
I guess what I meant more when I said magic number wasn't to say that the keyword values themselves were magic, but that they were magic within the context of our design system. For example, the fact that 800 corresponds to `as="strong" + weight="medium"` is just implied with the ternary operator instead of being explicitly defined.

Still going to change it back! Just thought I'd clarify. 😂]

Reply:
Not necessarily, since `Body` styles cascade still. A common use case of this component is to write docs etc. so, something like this is pretty common:
```
<Body as="p" weight="medium">
  <strong>All courses</strong> at MongoDB University are <em>free</em>!
</Body>
```

Reply:
I see. I envisioned the use case for Typography to feel more like the way [Mui does it](https://mui.com/components/typography/)

Reply:
As far as I can tell, you'd do it pretty similarly in Mui https://codesandbox.io/s/typographytheme-material-demo-forked-stpn8z?file=/index.js

Reply:
The browser default weight for `<strong>` is coming in that demo, `<Typography>` is setting the weight to 400; it doesn't try to interfere with anything in the children. It's generally fine because there shouldn't be any larger components inside a Body component, but setting top-level CSS here just felt icky.